### PR TITLE
feat: add typescript types

### PR DESCRIPTION
**Pull request by nfantone**
- feat: add new `index.d.ts` files with full `flynamo` client types.
- chore: remove `*.test.js` files from published artifact.
- updated all dependencies.
- chore: sort `package.json` keys.

> Types are only defined for the table-scoped `flynamo` client (i.e.: `forTable` return value).

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,4 +1,11 @@
+# Be aware that defininig patterns in "files" on package.json
```
Comment (crash7) [inline .npmignore:1]:
```suggestion
# Be aware that defining patterns in "files" on package.json
```

File: package.json [hunk 2]
```diff
@@ -39,34 +34,35 @@
...
+    "ramda": "^0.27.1"
   },
   "peerDependencies": {
     "aws-sdk": "2.x"
   },
   "devDependencies": {
-    "@commitlint/config-conventional": "^9.1.1",
-    "commitlint": "^9.1.0",
+    "@commitlint/config-conventional": "^11.0.0",
+    "commitlint": "^11.0.0",
```
Comment (crash7) [inline package.json:49]: Could you change this to the new package?

**Review by crash7 — APPROVED** [review]
Added two notes, but looks good. Thanks for doing this!

File: .npmignore (hunk 1 above)
Comment (nfantone) [reply .npmignore]: Removed `.npmignore` in favor of `"files"`.

File: package.json (hunk 2 above)
Comment (nfantone) [reply package.json]: Done.

File: package.json [hunk 3]
```diff
@@ -97,7 +100,7 @@
       [
         "@semantic-release/exec",
         {
-          "prepareCmd": "npx run-s generate-docs"
+          "prepareCmd": "npx run-s docs"
```
Comment (nfantone) [inline package.json:103]: Not sure why the `exec` script in run this way through `npx run-s` vs. using `yarn` (are dependencies not installed before it? Is `yarn` not available? Are paths solved differently?). In any case, I left it as is fearing it'd break CI if I changed it.

Comment (nfantone) [reply package.json:103]: Same applies to [`ramda-land`, BTW](https://github.com/flybondi/ramda-land/pull/20/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R130).

**Pull request merged**